proposals/0327-actor-initializers.md

```diff
@@ -211,343 +211,593 @@ struct S {
   }
+}
+```
 
-  // warning: nonisolated on a synchronous non-delegating initializer is redundant
-  nonisolated init(bad3: Void) {
-    self.friends = []
-    self.notifyAll()  // ❌ disallowed by escaping-use restriction.
-  }
+Once gaining access to an executor by crossing the first suspension point, crossing another suspension point does not change the executor, nor will that actually perform a suspension. Avoiding these unnecessary executor hops is an optimization that is done throughout Swift (e.g., self-recursive `async` and `isolated` functions).
+
+
+
+##### Initializers with `nonisolated self`
 
-  nonisolated init(ok: Void) async {
-    self.friends = []
-    self.notifyAll()  // ❌ disallowed by escaping-use restriction.
+The category of actor initializers that have a `nonisolated self` contain those which are non-async, or have an isolation that differs from being isolated to `self`. Unlike its methods, an actor's non-async initializer does _not_ require an `await` to be invoked, because there is no actor-instance to synchronize with. In addition, an initializer with a `nonisolated self` can access the instance's stored properties without synchronization, when it is safe to do so.
+
+Accesses to the stored properties of `self` is required to bootstrap an instance of an actor. Such accesses are considered to be a weaker form of isolation that relies on having exclusive access to the reference `self`. If `self` escapes the initializer, such uniqueness can no-longer be guaranteed without time-consuming analysis. Thus, the isolation of `self` decays (or changes) to `nonisolated` during any use of `self` that is not a direct stored-property access.  That change happens once on a given control-flow path and persists through the end of the initializer. Here are some example uses of `self` within an initializer that cause it to decay to a `nonisolated` reference:
+
+1. Passing `self` as an argument in a procedure call. This includes:
+  - Invoking a method of `self`.
+  - Accessing a computed property of `self`.
+2. Capturing `self` in a closure.
+3. Storing `self` to memory.
+
+Consider the following example that helps demonstrate how this isolation decay works:
+
+```swift
+class NotSendableString { /* ... */ }
+class Address: Sendable { /* ... */ }
+func greetCharlie(_ charlie: Charlie) {}
+
+actor Charlie {
+  var score: Int
+  let fixedNonSendable: NotSendableString
+  let fixedSendable: Address
+  var me: Self? = nil
+
+  func incrementScore() { self.score += 1 }
+  nonisolated func nonisolatedMethod() {}
+
+  init(_ initialScore: Int) {
+    self.score = initialScore
+    self.fixedNonSendable = NotSendableString("Charlie")
+    self.fixedSendable = NotSendableString("123 Main St.")
+
+    if score > 50 {
+      nonisolatedMethod() // a nonisolated use of `self`
+      greetCharlie(self)  // a nonisolated use of `self`
+      self.me = self      // a nonisolated use of `self`
+    } else if score < 50 {
+      score = 50
+    }
+ 
```

Comment:
So also this:

```
self.incrementScore() // ❌ error: 
```

right? Worth calling out since that's a big deal.

Reply:
Yep it's worth pointing out that isolated methods are not allowed anywhere.